### ISSUE TITLE
Modify way of how use local image as base image

### ISSUE THREAD

**Issue by marc-** [chef/chef-provisioning-docker#27]
I was trying to run sequentially creation of two images following way:
```ruby
require "chef/provisioning/docker_driver"
with_driver "docker"
machine_image "machine1" do
  recipe "my_recipe::1"
  machine_options :docker_options => {
      :base_image => {
          :name => "ubuntu",
          :repository => "ubuntu",
          :tag => "14.04"
      }
  }
end

machine_image "machine2" do
  recipe "my_recipe::2"

  machine_options :docker_options => {
      :base_image => {
          :name => "chef",
          :repository => "chef",
          :tag => "machine1"
       }
  }
end
``` 
But it doesn't work for some reason. Still looking on it.

**Comment by marc-** [comment]
There is actually `:from_image` docker option. Unfortunately it doesn't allow to preserve initial image repository name and tag.

**Comment by marc-** [comment]
How about remove following part:
```ruby
# Don't do this if we're loading from an image
if docker_options[:from_image]
"#{source_repository}:#{source_tag}"
else
```
and keep `:from_image` only to be able skip convergence? d961482 will allow to find specified image more flexible way.

**Comment by jkeiser** [comment]
@marc- you should be able to do this:

```ruby
require "chef/provisioning/docker_driver"
with_driver "docker"
machine_image "machine1" do
  recipe "my_recipe::1"
  machine_options :docker_options => {
      :base_image => {
          :name => "ubuntu",
          :repository => "ubuntu",
          :tag => "14.04"
      }
  }
end

machine_image "machine2" do
  recipe "my_recipe::2"
  from_image 'machine1'
end
```

**Comment by marc-** [comment]
This will work for this particular case. But what if you need to run machine from local image, wich wasn't created with `machine_image`, for instance built from Dockerfile?

**Comment by marc-** [comment]
Modifying my example to make it more clear:
```ruby
require "chef/provisioning/docker_driver"
with_driver "docker"

machine_image "machine" do
  recipe "my_cookbook::some_recipe"

  machine_options :docker_options => {
      :base_image => {
          :name => "image_built_from_dockerfile",
          :repository => "image_built_from_dockerfile",
          :tag => "latest"
       }
  }
end
```

**Comment by tyler-ball** [comment]
I'm just getting up to speed with all this, so I may be wrong.  But my initial analysis is that we should support base images from an existing docker image in a remote repository, or a local repository (from a Dockerfile) or another `machine_image` resource.  I think its confusing to have two different ways to specify a base image - 1) via the `from_image` attribute or 2) from the `:base_image` hash in the `:docker_options`.

Because `from_image` is a base `machine_options` attribute can we update it to support all the desired behaviors?  We could have it accept a string like `name:repository:tag` (because it only accepts a String now and not a Hash).  First, we try to find a resource named `name:repository:tag`.  If we cannot find that, look for an image using the `:name`, `:repository` and `:tag` parsed out from the String.

If we want to keep them separate, I think we should get much better documentation about the difference between the `from_image` attribute (can only reference other resources) and the `:base_image` hash in the `machine_options` attribute (references either a local or remote Docker repo).

\cc @jkeiser

**Comment by marc-** [comment]
Not sure if it's good idea to put docker specific options into chef-provisioning specific context.
Also, talking about refactoring and confusion, as there any justification for `:name` base image option?

**Comment by randomcamel** [comment]
The original issue should be fixed by PR #47. @tyler-ball go ahead and close, or leave open for discussion, as desired.

**Comment by randomcamel** [comment]
Reading more carefully I see that Tyler asked me to close it, so. :smile:
